m3-libs/libm3/src/arith/POSIX/MathPosix.m3

```diff
@@ -0,0 +1,37 @@
+MODULE MathPosix EXPORTS Math;
+
+(*
+  Glue to fix broken prototypes in CM3-d5.11.9
+
+  Author : Mika Nystrom <mika.nystroem@intel.com>
+  June, 2024
+
+  Copyright (c) Intel Corporation, 2024.  All rights reserved.
+
+  See COPYRIGHT-INTEL for terms.  
+
+  SPDX-License-Identifier: BSD-3-Clause
+
+ *)
+
+IMPORT MathPosixC;
+FROM Ctypes IMPORT int;
+
+PROCEDURE frexp (x: LONGREAL;  VAR exp: int): LONGREAL =
```

Comment:
I don't think you need an intermediate Modula-3 layer here.
You should just have a C layer.